Heart/Character/Domain/Entities/LevelEntity.php

```diff
@@ -89,4 +89,17 @@ public function getExperience(): int
     {
         return $this->experience;
     }
+
+    public function getPercentageExperience(int $currentExperience, int $currentLevel): float
+    {
+        $levelMin = $currentLevel * 89;
+
+        $difference = $currentExperience - $levelMin;
+
+        if ($difference == 0) {
+            return 100.0;
+        }
+
+        return floor($difference / (100 * 89) * 100);
+    }
```

Comment:
Por quê não usar os próprios métodos da classe? Não vejo motivo pra chamá-los 2x passando por referência.
```suggestion
    public function getPercentageExperience(): float
    {
        $levelMin = $this->getLevel() * 89;

        $difference = $this->getExperience() - $levelMin;

        if ($difference == 0) {
            return 100.0;
        }

        return floor($difference / (100 * 89) * 100);
    }
```